Source/ForecastDeskbarView.cpp

```diff
@@ -48,17 +52,15 @@ ForecastDeskbarView::AttachedToWindow()
 	AdoptParentColors();
 }
 
-extern "C" _EXPORT BView* instantiate_deskbar_item();
-
 
-BView*
+extern "C" _EXPORT BView*
 instantiate_deskbar_item()
```

Comment:
There's a `instantiate_deskbar_item(float maxWidth, float maxHeight)` that avoids hard-coding the icon size. The Deskbar tray and its icons are resized with the set system font size. Compare with, e.g. https://cgit.haiku-os.org/haiku/tree/src/apps/powerstatus/PowerStatusView.cpp#n929

Reply:
That makes sense, I'm trying to adapt the code but I'm struggling with [this part](https://github.com/dospuntos/Weather/blob/cb695af6f381da306371c731e661ce2f45786cc6/Source/ForecastDeskbarView.cpp#L91C2-L91C9):

```
BArchivable*
ForecastDeskbarView::Instantiate(BMessage* archive)
{
	if (!validate_instantiation(archive, "ForecastDeskbarView"))
		return NULL;

	return reinterpret_cast<BArchivable*>(instantiate_deskbar_item());
}
```
How can I include maxWidth and maxHeight in `instantiate_deskbar_item()`?

Reply:
Keep in mind that I don't really know what I'm doing, esp. when it comes to replicants...
The Instantiate() should return a ForecastDeskbarView*, not a BArchivable*.

Reply:
> Keep in mind that I don't really know what I'm doing, esp. when it comes to replicants... 

Hard to believe...

> The Instantiate() should return a ForecastDeskbarView*, not a BArchivable*.

I noticed that from the examples, however with the current code this caused the app to crash. As it is now (with my latest update), the code works and the icon will resize according to the system font size.